lib/chef/provider/git.rb

```diff
@@ -301,7 +301,16 @@ def run_options(run_opts = {})
           # user who is executing `git` not the user running Chef.
           env["HOME"] = begin
             require "etc"
-            Etc.getpwnam(@new_resource.user).dir
+            case @new_resource.user
```

Comment:
I see, because the resource type is already constrained in https://github.com/chef/chef/blob/master/lib/chef/resource/scm.rb#L72, there is no reason it would go into the else case right?